File: python/Reverse_String.py
"""
Write a function that takes a string as input and returns the string reversed.

Example:
Given s = "hello", return "olleh".
"""

class Solution(object):
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        return s[::-1]

    def reverseString2(self, s):
        """
        :type s: str
        :rtype: str
        """
        str = list(s)
        left, right = 0, len(str) - 1
        while left < right:
            str[left], str[right] = str[right], str[left]
            left += 1
            right -= 1
        return "".join(str)

File: cpp/Reverse_String.cc
/*
Write a function that takes a string as input and returns the string reversed.

Example:
Given s = "hello", return "olleh".
*/

class Solution
{
public:
    string reverseString(string s)
    {
        if (s.size() == 0)
            return s;

        int left = 0, right = s.size() - 1;
        while (left < right)
        {
            char tmp = s[left];
            s[left] = s[right];
            s[right] = tmp;
            left++;
            right--;
        }

        return s;
    }
};
